sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -585,17 +591,6 @@ static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
   // Clean custom headers previously appended
   curl_slist_free_all(p_list);
 
-  // make sure to set the end of the body response as the end of the complete response
-  if (az_succeeded(result))
```

Comment:
Can you please explain why we don't need to append a `0` at the end anymore?

Reply:
There is not a reason why to add it or why to not add it really.
It was just a way to let customers use printf(...) directly,  but they can still use printf("%.*s", size, prt)..  So, We don't really need to provide this or as other transport implementations to do it.

And at any case, it wouldn't be the transport http adapter who would add this, but the pipeline or some other layer on top.. So it works for any transport adapter below it.